### PR TITLE
docs: update tuner docs

### PR DESCRIPTION
**Pull request by tadejsv**

### CODE REVIEW

File: docs/components/tuner/loss.md [hunk 1]
```diff
@@ -65,20 +65,21 @@ tuner = PaddleTuner(..., loss=loss)
...
 
-The image below illustrates the relative position of samples in the case of triplet training and will help us to define the necessary terminology. 
+```{figure} sample-position-during-mining.png
+:width: 50%
+```
+
+The white circle in the middle (`A`) denotes our anchor sample $\mathrm{x}_i$ and the white circle on the left (`P`) denotes our positive sample $\mathrm{x}_p$. Depending on where our negative sample will fall, inside the encoding space, it will be either *hard*, *semihard* or an *easy* sample. More precisely, a negative sample is considered to be
 
-![Miner sample positions](../../img/sample-position-during-mining.png "Sample Positions in Mining")
+- **hard**, when it its distance to the anchor sample is smaller than the distance between the anchor and the positive sample, formally $d(\mathrm{x}_i,\mathrm{x}_n) < d(\mathrm{x}_i,\mathrm{x}_p)$.
```
Comment (gvondulong) [inline docs/components/tuner/loss.md:76]: `It its`

File: docs/components/tuner/loss.md [hunk 2]
```diff
@@ -65,20 +65,21 @@ tuner = PaddleTuner(..., loss=loss)
...
-When a negative sample is just *a little* further from the postive sample $\mathrm{x}_i$, but within a $\mathrm{margin}$, the sample is considered to be *semihard*. 
-Formally: $d(\mathrm{x}_i,\mathrm{x}_n) < d(\mathrm{x}_i,\mathrm{x}_p) + \mathrm{margin}$. 
-Any negative sample that is even further away from the anchor, is considered an *easy* sample, i. e. $d(\mathrm{x}_i,\mathrm{x}_n) > d(\mathrm{x}_i,\mathrm{x}_p)$.
-It is possible to apply each of the strategies on both the positive and the negative samples. 
-This means that, when given an anchor and a negative sample, we can freely specify the strategy used to retrieve the positive sample for the same anchor. 
+The strategies presented here were given for the negative sample, taking the positive one as given - but analogous strategies for selecting the positive sample, having the negative one already give, are also possible. We enable the use of both positive and negative mining strategy in Tuner.
```
Comment (gvondulong) [inline docs/components/tuner/loss.md:82]: Possibly break up the sentence at the dash. It is quite long.

File: docs/components/tuner/callbacks.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Callbacks
+
+Callbacks offer a way to integrate various auxiliary tasks into the training loop. We already offer built-in callbacks for some common tasks, such as
```
Comment (gvondulong) [inline docs/components/tuner/callbacks.md:3]: I would remove `already` from the latter sentence. That might make the sentence age better. If Jina exists for 5 years, the features will have outgrown the mention `already` here.

File: docs/components/tuner/callbacks.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+You should then be able to see your training runs in wandb.
+
+## Custom callbacks
+
+If the existing callbacks don't provide the functionality you need, you can easily write your own.
+
+To do that, you should subclass {class}`~finetuner.tuner.callback.base.BaseCallback`, and override the `on_*` methods that you need. Each method will get a `tuner` object as an argument - this gives you access to its `.state` attribute, which is a {class}`~finetuner.tuner.state.TunerState` object, and contains useful information on the current training run.
+
+Here's a simple example of a callback that time each training epoch, and prints the elapsed time at the end
```
Comment (gvondulong) [inline docs/components/tuner/callbacks.md:41]: time`s` each training epoch.

Comment (gvondulong) [reply docs/components/tuner/callbacks.md:41]: Full stop is missing in the end.

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Comment by gvondulong** [comment]
Great changes! Looks good to me!

File: docs/components/tuner/callbacks.md (hunk 4 above)
Comment (tadejsv) [reply docs/components/tuner/callbacks.md:41]: @gvondulong for such changes next time use the inline edit tool - it will be easier for me to commit them

Comment (tadejsv) [reply docs/components/tuner/callbacks.md:41]: Screenshot from 2021-12-16 11-51-33

**Review by gvondulong — APPROVED** [review]
Good to go.

**Pull request merged**